Tfm crypto modules update

joerchan opened the pull request:
Update TF-M Crypto module Kconfig entries to match current version of TF-M.

GENERATOR was split across KEY and KEY_DERIVATION.
ASSYMETRIC was split into ASYM_ENCRYPT and ASYM_SIGN.

joerchan committed:
modules: tfm: Remove CRYPTO_GENERATOR_MODULE kconfig

Remove CRYPTO_GENERATOR_MODULE Kconfig which has been removed.
Its features are covered by TFM_CRYPTO_KEY_MODULE and
CRYPTO_KEY_DERIVATION_MODULE.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan committed:
modules: tfm: Split CRYPTO_ASYMMETRIC module into SIGN and ENCRYPT

The ASSYMETRIC crypto module has been split in two, for sign and
encrypt. Update Kconfig entries to match.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

microbuilder commented:
Is the CI check here related? It's on an nRF board.

joerchan commented:
> Is the CI check here related? It's on an nRF board.

I really doubt it. It is failing in network tests, so I don't think TF-M is even enabled in those tests. I'm checking on discord in the infrastructure channel.

microbuilder commented:
> > Is the CI check here related? It's on an nRF board.
> 
> 
> 
> I really doubt it. It is failing in network tests, so I don't think TF-M is even enabled in those tests. I'm checking on discord in the infrastructure channel.

I just thought of this PR which was recently merged to resolve a conflict with TFM and mbedtls both being available at the same time on the NS side, and thought the TLS failure here may be related. https://github.com/zephyrproject-rtos/mbedtls/pull/33

tejlmand commented on modules/trusted-firmware-m/Kconfig.tfm.crypto_modules:
```diff
@@ -54,12 +54,20 @@ config TFM_CRYPTO_CIPHER_MODULE_ENABLED
-config TFM_CRYPTO_ASYMMETRIC_MODULE_ENABLED
-	bool "Enable ASYMMETRIC crypto module"
+config TFM_CRYPTO_ASYM_ENCRYPT_MODULE_ENABLED
+	bool "Enable ASYM ENCRYPT crypto module"
```
As per the guidelines, please remove `Enable`:
https://docs.zephyrproject.org/latest/guides/build/kconfig/tips.html#prompt-strings

> For a Kconfig symbol that enables a driver/subsystem FOO, consider having just “Foo” as the prompt, instead of “Enable Foo support” or the like.

tejlmand commented on modules/trusted-firmware-m/Kconfig.tfm.crypto_modules:
```diff
@@ -54,12 +54,20 @@ config TFM_CRYPTO_CIPHER_MODULE_ENABLED
+	  is not used.
+
+config TFM_CRYPTO_ASYM_SIGN_MODULE_ENABLED
+	bool "Enable ASYM SIGN crypto module"
```
As per the guidelines, please remove `Enable`:
https://docs.zephyrproject.org/latest/guides/build/kconfig/tips.html#prompt-strings

> For a Kconfig symbol that enables a driver/subsystem FOO, consider having just “Foo” as the prompt, instead of “Enable Foo support” or the like.

tejlmand reviewed (commented):
Minor nits.

joerchan committed:
modules: tfm: Update crypto modules prompt strings

Update crypto modules prompt strings to follow the Kconfig guidelines.
"For a Kconfig symbol that enables a driver/subsystem FOO, consider
having just “Foo” as the prompt, instead of “Enable Foo support” or the
like."

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan commented:
@tejlmand Added another commit to update the prompt message.

frkv reviewed (approved):
LGTM, pending passing tests

tejlmand reviewed (approved):
lgtm
